python/tank/templatekey.py

```diff
@@ -328,7 +329,22 @@ def __init__(self,
         elif self._filter_by is not None:
             # filter_by is a regex
             self._custom_regex_u = re.compile(self._filter_by, re.UNICODE)
-        
+
+        self._subset_str = subset
+        self._subset_format = subset_format
+        if self._subset_format and sys.version_info < (2, 6):
+            raise TankError("Subset formatting in template keys require python 2.6+!")
+
+        self._subset_str = subset
+
+        if subset:
+            try:
+                self._subset_regex = re.compile(subset, re.UNICODE)
+            except Exception, e:
```

Comment:
Doing `except Exception as e:` will result in a compile-time error, e.g even if the code is never executed in python2.5, it will still not load.

Did a quick test with this code and it doesn't run in py25:

```python
if 0:
  try:
     print "test"
  except Exception as e:
     print e
```